src/main/scala/scalaz/reactive/Future.scala

```diff
@@ -2,33 +2,44 @@ package scalaz.reactive
 
 import scalaz.Scalaz._
 import scalaz.reactive.Time.{ NegInf, PosInf }
+import scalaz.zio.IO
 import scalaz.{ Applicative, Functor, Monad, Monoid }
 
-case class Future[+A](time: () => Time, force: () => A) {
+object Types {
+  type Infallible[A] = IO[Nothing, A]
```

Comment:
Shouldn't this be `type Infallible[+A] = IO[Void, +A]`? to ensure that this does not get flat mapped with an IO which does fail? or maybe I'm wrong